lib/puppet/provider/package/gem.rb

```diff
@@ -121,13 +127,18 @@ def install(useversion = true)
           command << source
         else
           # interpret it as a gem repository
-          command << "--source" << "#{source}" << resource[:name]
+          command << "--source" << "#{source}"
```

Comment:
This looks unrelated to the `--no-document` issue?

Reply:
The `if source = resource[:source]` statement should be limited to adding `source` to the `command` array. We should always add `--no-rdoc ...` to the `command` array to never install documentation: independent of the whether or not a :source parameter is defined. They are absolutely unrelated.